test/new-e2e/examples/vm_with_file_operations_test.go

```diff
@@ -22,10 +25,77 @@ type vmSuiteWithFileOperations struct {
 
 // TestVMSuiteWithFileOperations runs tests for the VM interface to ensure its implementation is correct.
 func TestVMSuiteWithFileOperations(t *testing.T) {
-	suiteParams := []e2e.SuiteOption{e2e.WithProvisioner(awshost.ProvisionerNoFakeIntake(awshost.WithEC2InstanceOptions(ec2.WithOS(os.WindowsDefault))))}
+	suiteParams := []e2e.SuiteOption{e2e.WithProvisioner(awshost.ProvisionerNoAgentNoFakeIntake(awshost.WithEC2InstanceOptions(ec2.WithOS(os.WindowsDefault))))}
 	e2e.Run(t, &vmSuiteWithFileOperations{}, suiteParams...)
 }
 
+func assertExitCodeEqual(t *testing.T, err error, expected int, msgAndArgs ...interface{}) {
+	t.Helper()
+	var exitErr *ssh.ExitError
+	assert.ErrorAs(t, err, &exitErr)
+	assert.Equal(t, expected, exitErr.ExitStatus(), msgAndArgs)
+}
+
+// TestCommandResults tests that commands return output or errors in the expected way
+func (v *vmSuiteWithFileOperations) TestCommandResults() {
+	vm := v.Env().RemoteHost
+
+	// successful command should return the output
+	out, err := vm.Execute("echo hello")
+	v.Assert().NoError(err)
+	v.Assert().Contains(out, "hello")
+
+	// invalid commands should return an error
+	_, err = vm.Execute("not-a-command")
+	v.Assert().Error(err, "invalid command should return an error")
+
+	// specific exit code should be returned
+	_, err = vm.Execute("exit 2")
+	v.Assert().Error(err, "non-zero exit code should return an error")
+	assertExitCodeEqual(v.T(), err, 2, "specific exit code should be returned")
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement. -->
**🥜 nitpick**
Would rather have the assertion explicit, as in 
```suggestion
	assert.Equal(t, expected, exitErr.ExitStatus(), msgAndArgs)
```

or move the helper closer to `host.go`. As is I consider it hard to discover that the error returned by `Execute` is an `ssh.Error`. Should we have `ExecuteWithExitCode` instead, or `Execute` returning `output, exitCode, error`?

Reply:
I would keep everything here - I know golang is sometimes verbose, but this helps readability in this example:
```suggestion
	var exitErr *ssh.ExitError
	require.ErrorAs(v.T(), err, &exitErr, "error code should contain ssh.ExitError in its chain")
	assert.Equal(v.T(), exitErr.ExitStatus(), 2, "specific exit code should be returned")
```